zgw/documenten-api/src/main/kotlin/com/ritense/documentenapi/service/DocumentenApiColumnDeploymentService.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2015-2024 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.documentenapi.service
+
+import com.ritense.authorization.annotation.RunWithoutAuthorization
+import com.ritense.document.domain.event.DocumentDefinitionDeployedEvent
+import com.ritense.documentenapi.domain.DocumentenApiColumn
+import com.ritense.documentenapi.domain.DocumentenApiColumnId
+import com.ritense.documentenapi.domain.DocumentenApiColumnKey.AUTEUR
+import com.ritense.documentenapi.domain.DocumentenApiColumnKey.BESTANDSOMVANG
+import com.ritense.documentenapi.domain.DocumentenApiColumnKey.CREATIEDATUM
+import com.ritense.documentenapi.domain.DocumentenApiColumnKey.INFORMATIEOBJECTTYPE
+import com.ritense.documentenapi.domain.DocumentenApiColumnKey.TITEL
+import org.springframework.context.event.EventListener
+import org.springframework.transaction.annotation.Transactional
+
+open class DocumentenApiColumnDeploymentService(
+    private val documentenApiService: DocumentenApiService,
+) {
+
+    @Transactional
+    @RunWithoutAuthorization
+    @EventListener(DocumentDefinitionDeployedEvent::class)
+    open fun createCaseTabs(event: DocumentDefinitionDeployedEvent) {
+        if (event.documentDefinition().id().version() == 1L) {
```

Comment:
Shouldn't this only be done if there's no configuration for the columns available? If so, then the `DocumentDefinitionDeployedEvent` might not be the right trigger for this

Reply:
> Shouldn't this only be done if there's no configuration for the columns available?
No. It should only be done when the definition is deployed for the first time

Reply:
I should have made that clearer. I meant:

Shouldn't this only be done if there's no configuration for the columns available and it's the deployment is deployed for the first time?

Reply:
A case can't have configuration if it doesn't exist